client/extensions/woocommerce/state/action-list/README.md

```diff
@@ -20,107 +20,115 @@ created.)
-step, and then the step object is given a `startTime` field with a timestamp.
-This both tracks timing and denotes the current status of the step as "running".
-
-#### Action
-
-This part is simple. When the step is started, the associated action is dispatched.
-
-#### Asynchronous Response
+As the first step starts, it is designated as the `currentStep`
+```
+{
+  currentStep: step1,
+  nextSteps: [ step2, step3, step4, step5 ],
+}
+```
 
-For asynchronous operations, it's recommended that a data-layer handler be set to
-listen for step actions and perform the asynchronous steps as necessary.
+Then after it completes, it moves to the `prevSteps` array.
+```
+{
+  prevSteps: [ step1 ],
+  currentStep: null,
+  nextSteps: [ step2, step3, step4, step5 ],
+}
+```
 
-It's the responsibility of such a handler to ensure that the appropriate success
-or failure action is dispatched at the conclusion of the asynchronous operation.
+Then step2 is started.
+```
+{
+  prevSteps: [ step1 ],
+  currentStep: step2,
+  nextSteps: [ step3, step4, step5 ],
+}
+```
 
-This should map easily to any sort of promise situation by having the `then` dispatch
-success, and the `catch` dispatch failure.
+And so on...
+```
+{
+  prevSteps: [ step1, step2 ],
+  currentStep: null,
+  nextSteps: [ step3, step4, step5 ],
+}
+```
 
-Assuming the action has a follow-up asynchronous response, it will be handled based on status:
+Until all steps are complete.
+```
+{
+  prevSteps: [ step1, step2, step3, step4, step5 ],
+  currentStep: null,
+  nextSteps: [],
+}
+```
 
-#### Edit State Updates
+### Step objects
 
-It is also the responsibility of the handlers to ensure that the edit state is updated
-appropriately after each step. This means if an API object was created, the state
-should be updated accordingly, for example.
+Each step object consists initially of a description and an `onStep` event.
 
-#### Action List Updates
+```
+{ description: translate( 'Reticulating Splines' ), onStep: reticulateSplines }
+```
 
-For many cases, there will be placeholder ids for things that have not yet been created.
-It's important for each handler to update these placeholder ids with real ones as they
-become available each step along the way.
+Note that the description is designed for user display, so it should be translated.
 
-#### Step End
+The `onStep` function has two parameters that are passed to it:
+```
+function onStep( dispatch, actionList ) {}
+```
 
-The step is given an `endTime` field with a timestamp. This tracks timing for the
-operation as well as status of the step as "complete" if successful, or "failed" if
-an `error` field is present. If there are more steps after this one, the next
-step is started. If this is the last step, then no further action is taken.
+The `dispatch` function can be used to send off actions to the system. These can
+be thunks or asynchronous actions, anything that the store dispatch can handle.
```

Comment:
> thunks…anything the store dispatch can handle

probably best to omit this since it's only true because we have the thunk middleware injected. I like discouraging their use and part of that is removing them from example/README/documentation files

> asynchronous actions

in this context I don't see what this means. `actions` is a well-defined term in the `Redux` world and there is no such thing as an _asynchronous action_ there.